api/common/upgradeseries_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common_test
+
+import (
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/names.v2"
+
+	apitesting "github.com/juju/juju/api/base/testing"
+	"github.com/juju/juju/api/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/core/model"
+)
+
+type upgradeSeriesSuite struct {
+	testing.IsolationSuite
+	tag names.Tag
+}
+
+var _ = gc.Suite(&upgradeSeriesSuite{})
+
+func (s *upgradeSeriesSuite) SetUpTest(c *gc.C) {
+	s.tag = names.NewMachineTag("0")
+}
+
+func (s *upgradeSeriesSuite) TestWatchUpgradeSeriesNotifications(c *gc.C) {
+	facadeCaller := apitesting.StubFacadeCaller{Stub: &testing.Stub{}}
+	facadeCaller.FacadeCallFn = func(name string, args, response interface{}) error {
+		c.Assert(name, gc.Equals, "WatchUpgradeSeriesNotifications")
+		c.Assert(args, jc.DeepEquals, params.Entities{Entities: []params.Entity{
+			{Tag: s.tag.String()},
+		}})
+		*(response.(*params.NotifyWatchResults)) = params.NotifyWatchResults{
+			Results: []params.NotifyWatchResult{{
+				NotifyWatcherId: "1",
+				Error:           nil,
+			}},
+		}
+		return nil
+	}
+	apiCaller := apitesting.APICallerFunc(
+		func(objType string,
+			version int,
+			id, request string,
+			a, result interface{},
+		) error {
+			c.Check(objType, gc.Equals, "NotifyWatcher")
+			c.Check(id, gc.Equals, "1")
+			c.Check(request, gc.Equals, "Next")
+			c.Check(a, gc.IsNil)
+			return nil
+		},
+	)
+	facadeCaller.ReturnRawAPICaller = apitesting.BestVersionCaller{APICallerFunc: apiCaller, BestVersion: 1}
+
+	api := common.NewUpgradeSeriesAPI(&facadeCaller, s.tag)
+	_, err := api.WatchUpgradeSeriesNotifications()
```

Comment:
Not satisfied with this test.  The ReturnRawAPICaller is needed for the call to apiwatcher.NewNotifyWatcher.  Verifying the results is currently problematic.  Suggestions welcome.